jaeger-core/src/test/java/com/uber/jaeger/propagation/B3TextMapCodecResiliencyTest.java

```diff
@@ -66,4 +66,21 @@ private TextMap validHeaders() {
     maliciousCarrier.put(PARENT_SPAN_ID_NAME, validInput);
     return maliciousCarrier;
   }
+
+  @Test
+  public void shouldFallbackWhenExtractingFromFaultyTextMap() {
+    TextMap faultyTextMap = new FaultyTextMap();
+    //when
+    SpanContext extract = sut.extract(faultyTextMap);
+    //then
+    assertNull(extract);
+  }
+
+  @Test
+  public void shouldFallbackWhenInjectingIntoFaultyTextMap() {
+    TextMap faultyTextMap = new FaultyTextMap();
+    SpanContext spanContext = SpanContext.contextFromString("a:b:c:1");
+    //expect to pass
+    sut.inject(spanContext, faultyTextMap);
```

Comment:
Would it be possible to have a test to ensure the state of the `SpanContext` is sane after the exception? Mainly to address @vprithvi's comment https://github.com/jaegertracing/jaeger-client-java/pull/359/files#r171026709

Reply:
As I understand @vprithvi was concerned about polluting `TextMap` with partial data. Unfortunately, I don't see a way to prevent that as we don't have any "transactional" method to set all fields at once + there is no method to compensate the action.